android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java

```diff
@@ -376,16 +379,34 @@ private void downloadDeviceIcon(final String url){
 			@Override
 			public void run(){
 				try{
-					deviceLogo = AndroidTools.downloadImage(url);
+					if(mLockScreenDeviceIconManager.updateCachedImage(url)) {
+						Log.d(TAG, "URL: " + url);
+						Log.d(TAG, "Image Update Needed");
+						deviceLogo = AndroidTools.downloadImage(url);
+						mLockScreenDeviceIconManager.saveFileToCache(deviceLogo, url);
+					} else {
+						Log.d(TAG, "Image Is Up To Date");
```

Comment:
This doesn't always mean the image is up to date right? I believe the app will hit this else statement if the image is not cached at all. 
I suggest updating the log to be more clear or restructuring the if statements to separate the three cases to make it easier to follow:
1-image not cached
2-image cached and valid 
3-image cached but expired

Reply:
the method should only return false if the image is cached and valid, all other instances return true meaning we need to cache a new icon